common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt

```diff
@@ -51,9 +57,42 @@ class ContainerToolsRule : TestRule {
         ideaProjectTestFixture =
             IdeaTestFixtureFactory.getFixtureFactory().createLightFixtureBuilder().fixture
         EdtTestUtil.runInEdtAndWait(ThrowableRunnable { ideaProjectTestFixture.setUp() })
+
+        MockKAnnotations.init(testInstance, relaxUnitFun = true)
+        replaceServices()
     }
 
     private fun tearDownRule() {
         EdtTestUtil.runInEdtAndWait(ThrowableRunnable { ideaProjectTestFixture.tearDown() })
     }
+
+    /**
+     * Replaces all services annotated with [TestService].
+     */
+    private fun replaceServices() {
+        for (member in testInstance::class.declaredMemberProperties) {
+            if (member.annotations.filter { it is TestService }.isNotEmpty()) {
+                member as KProperty1<Any?, Any?>
+                member.isAccessible = true
+                val service: Any = member.get(testInstance)!!
+                setService(service)
+            }
+        }
+    }
+
+    /**
+     * Replaces the service binding in the [MutablePicoContainer] with the given instance and
+     * returns the original service instance.
+     *
+     * @param newInstance the new instance to register
+     */
+    private fun setService(newInstance: Any) {
+        val applicationContainer =
+            ApplicationManager.getApplication().picoContainer as MutablePicoContainer
```

Comment:
`with (Application...) { ... }` is more concise

Reply:
Thanks for the tip, done.